cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,449 @@
+        if (errorConfig.isNestedError) {
+          nestedErrors.push(err);
+        } else {
+          localErrors.push(err);
+        }
+      } catch (error) {
+        return;
+      }
+    });
+
+    const transforms = this.props.transforms;
+
+    // WidgetProps for inputFieldDropdown
+    const inputFieldProps = {
+      multiselect: true,
+      allowedTypes: [],
+    };
+
+    // WidgetProps for ConfiguartionGroup that will hold transform properties
+    const properties: PluginProperties = {};
+    const config: IConfigurationGroupProps = {
+      classes: {},
+      errors: {},
+      values: {},
+      pluginProperties: properties,
+    };
+
+    if (this.state.transform !== '') {
+      inputFieldProps.allowedTypes =
+        transforms.filter((trans) => trans.name === this.state.transform)[0].supportedTypes || [];
+
+      const transform: ITransformProp = transforms.filter(
+        (t) => t.name === this.state.transform
+      )[0];
+
+      // Populating the pluginProperties for each transform property
+      transform.options.forEach((transProp) => {
+        const pluginProp: IPluginProperty = {
+          name: transProp.name,
+        };
+        if (transProp['widget-attributes']) {
+          if (transProp['widget-attributes'].macro) {
+            pluginProp.macroSupported = transProp['widget-attributes'].macro;
+          }
+          if (transProp['widget-attributes'].description) {
+            pluginProp.description = transProp['widget-attributes'].description;
+          }
+          if (transProp['widget-attributes'].required) {
+            pluginProp.required = transProp['widget-attributes'].required;
+          }
+        }
+        properties[transProp.name] = pluginProp;
+      });
+
+      config.pluginProperties = properties;
+      // Parsing errors for ConfigurationGroup
+      config.errors = {};
+      nestedErrors.forEach((err) => {
+        try {
+          const errorConfig: IErrorConfig = JSON.parse(err.element);
+          const errorObj = { msg: err.msg };
+          if (errorConfig.transformPropertyId in config.errors) {
+            config.errors[errorConfig.transformPropertyId].push(errorObj);
+          } else {
+            config.errors[errorConfig.transformPropertyId] = [errorObj];
+          }
+        } catch (error) {
+          return;
+        }
+      });
+
+      config.widgetJson = {
+        'configuration-groups': [
+          {
+            label: transform.label + ' properties',
+            properties: transform.options,
+          },
+        ],
+        filters: transform.filters, // Passing ConfigurationGroup filters to allow for dynamic showing/hiding of properties
+      };
+      config.values = this.state.transformProperties;
+      config.classes = this.props.classes;
+      config.onChange = this.handleChangeTransformOptions.bind(this);
+
+      if (nestedErrors.length > 0) {
+        this.handleChange('expanded', true);
+      }
+    }
```

Comment:
it will be good to extract this to a utility function and write unit test for it

Reply:
Are you referring just the nestErrors check or the whole logic that creates the props for the ConfigurationGroup?

Reply:
Done